spec/hyrax/transactions/steps/save_spec.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+require 'spec_helper'
+require 'hyrax/transactions/steps/save'
+require 'hyrax/specs/spy_listener'
+
+RSpec.describe Hyrax::Transactions::Steps::Save do
+  subject(:step)      { described_class.new(persister: persister) }
+  let(:adapter)       { Valkyrie::MetadataAdapter.find(:test_adapter) }
+  let(:change_set)    { change_set_class.new(resource) }
+  let(:persister)     { adapter.persister }
+  let(:resource)      { build(:hyrax_work) }
+  let(:query_service) { adapter.query_service }
+  let(:listener)      { Hyrax::Specs::SpyListener.new }
+
+  before { Hyrax.publisher.subscribe(listener) }
+  after  { Hyrax.publisher.unsubscribe(listener) }
+
+  let(:change_set_class) do
+    Class.new(Hyrax::ChangeSet) { self.fields = [:title] }
+  end
+
+  describe '#call' do
+    it 'is a success' do
```

Comment:
I believe a wrapping context for lines 23 to 36 for "when the save succeeds" (or some such thing).

Reply:
i tend to have a practice of leaving some default case ("happy path"?) outside a context in specs. i'm happy to change these ones in particular on your say-so, but it might be worth considering the desired convention across the code base